### DISCUSSION TITLE
Run experiments offline on an iPad

### DISCUSSION THREAD

**Discussion by steffejr** [jspsych/jsPsych#3021] [General]
Hello. I have developed a series of tasks with jsPsych. I would like to administer them with a tablet (iPad) since many of the tasks are drawing tasks, e.g. cube copy, trail making, etc. A current study requires that all data collection is done offline. 
The tasks run fine offline on a laptop but I would like to run them offline on an iPad. Does anyone have any suggestions?

I have been able to git clone by code to the iPad and access it in Files using "ish." I have tried to open the files in Chrome/Safari/Edge ... no luck.

I have tried using ish to run a python webserver. This has had some success but is not reliable. It is really slow and cannot play sounds!
I have tried to use "worldwideweb" app as a server to run the code. This also has some unreliable success. The server keeps stopping and will not play sounds.

Does anyone have any ideas or experience with this?
Thank you

**Comment by georginatorok** [comment]
Hey,
I'm not sure if this will be any helpful but perhaps will give you some pointers in a way. I've run jspsych tasks locally on a Samsung tablet using the Fully Kiosk Browser app, which bypasses the strict privacy measures due to which my html files could not be opened in Chrome. Data collection in the app works fine. 
As far as I know the Fully Kiosk app is Android-only but maybe if you find a similar kiosk app that runs on an iPad you could run your tasks through that. Maybe something like this? https://scalefusion.com/ios-kiosk-browser 
Hope you find a solution (if you do, please let us know here because I would also be interested to know how we could extend our testing to iPads too).
Gina
